src/hazardous/base.rs

```diff
@@ -0,0 +1,542 @@
+//! # Base
+//!
+//! This module provides convenient, type-parameterized byte-storage types
+//! that can be used to quickly create newtypes for keys, digests, etc.
+//! The goal is to support all the various cryptography types that are
+//! basically just bags of bytes, but **absolutely cannot** afford to
+//! be confused with one another.
+//!
+//! To that end, we define two structs — `PublicData` and `SecretData` —
+//! to act as generic containers for public and secret information, respectively.
+//! These containers are each parameterized by the same two type parameters:
+//! `B` (for "bytes") and `C` (for "context").
+//!
+//! ## Parameter: `B` (bytes)
+//! `B` parameterizes over the **byte storage**. In practice, this is
+//! either an [`ArrayData`][a] or [`VecData`][b]. This allows us
+//! to implement methods on any type that can be converted from
```

Comment:
I think this line of the docs is confusing. I'll change it in the next iteration.